sw/device/silicon_creator/lib/drivers/address_translation.h

```diff
@@ -0,0 +1,61 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_ADDRESS_TRANSLATION_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_ADDRESS_TRANSLATION_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * TODO: doc
+ *
+ */
+typedef enum address_translation_slot {
+  address_translation_slot_0,
+  address_translation_slot_1,
+  address_translation_slot_count,
+} address_translation_slot_t;
+
+/**
+ * TODO: doc
+ *
+ */
+typedef struct address_translation_pair {
+  uintptr_t matching_addr;
+  uintptr_t remap_addr;
+  size_t size;
+} address_translation_pair_t;
+
+/**
+ * TODO: doc
+ *
+ */
+typedef struct address_translation_region {
```

Comment:
I can't think of a situation where we'd want to configure different I and D regions in silicon creator firmware. Maybe we can just get away with a single address translation pair.

Reply:
Done